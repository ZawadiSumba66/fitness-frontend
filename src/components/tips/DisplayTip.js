import { Link } from '@reach/router';
import PropTypes from 'prop-types';

const DisplayTip = ({ title, description, id }) => (
  <div className="shadow rounded infoCarousel__wrapper-desc-info mx-auto mb-5">
    <h3 className="text-center font-weight-bold test-class">{title}</h3>
    <div className="bg-white align-items-center p-3">
      <p>{description}</p>
    </div>
    <Link to={`/tips/${id}`} className="button-orange p-2 ml-3 text-light font-weight-bold">
      View more details
    </Link>
    <div className="pt-4 px-3">
      <span className="fa fa-star checked" />
      <span className="fa fa-star checked" />
      <span className="fa fa-star checked" />
      <span className="fa fa-star checked" />
      <span className="fa fa-star text-sm-gray" />
    </div>
  </div>
);
DisplayTip.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};
export default DisplayTip;
