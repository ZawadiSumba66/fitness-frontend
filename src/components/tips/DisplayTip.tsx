import { Link } from '@reach/router';

type TipDisplay = {
  title: string,
  description: string,
  id: number,
};

const DisplayTip = ({
  title, description, id,
}: TipDisplay) => (
  <div className="shadow rounded infoCarousel__wrapper-desc-info mx-auto mb-5">
    <h3 className="text-center font-bold text-3xl  test-class">{title}</h3>
    <div className="bg-white align-items-center p-3">
      <p>{description}</p>
    </div>
    <Link to={`/tips/${id}`} className="button-orange p-2 ml-3 text-light font-weight-bold">
      View more details
    </Link>
    <div className="d-flex pt-12">
      <div className="px-3">
        <span className="fa fa-star checked" />
        <span className="fa fa-star checked" />
        <span className="fa fa-star checked" />
        <span className="fa fa-star checked" />
        <span className="fa fa-star text-sm-gray" />
      </div>
    </div>
  </div>
);

export default DisplayTip;
