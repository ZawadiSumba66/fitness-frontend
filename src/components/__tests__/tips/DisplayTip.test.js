import renderer from 'react-test-renderer';
import DisplayTip from '../../tips/DisplayTip';

it('matches the Display Tip snapshot', () => {
  const tree = renderer.create(<DisplayTip />).toJSON();
  expect(tree).toMatchSnapshot();
});
