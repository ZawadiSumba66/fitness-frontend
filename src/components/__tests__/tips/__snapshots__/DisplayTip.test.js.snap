// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Display Tip snapshot 1`] = `
<div
  className="shadow rounded infoCarousel__wrapper-desc-info mx-auto mb-5"
>
  <h3
    className="text-center font-weight-bold test-class"
  />
  <div
    className="bg-white align-items-center p-3"
  >
    <p />
  </div>
  <a
    className="button-orange p-2 ml-3 text-light font-weight-bold"
    href="/tips/undefined"
    onClick={[Function]}
  >
    View more details
  </a>
  <div
    className="pt-4 px-3"
  >
    <span
      className="fa fa-star checked"
    />
    <span
      className="fa fa-star checked"
    />
    <span
      className="fa fa-star checked"
    />
    <span
      className="fa fa-star checked"
    />
    <span
      className="fa fa-star text-sm-gray"
    />
  </div>
</div>
`;
